packages/eslint-plugin/src/rules/consistent-callback-type/index.test.js

```diff
@@ -0,0 +1,67 @@
+// @ts-nocheck
+const { ESLintUtils } = require('@typescript-eslint/experimental-utils');
+const rule = require('./index');
+
+const ruleTester = new ESLintUtils.RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('consistent-callback-type', rule, {
+  valid: [
+    // Valid when prop is TSTypeAliasDeclaration and the callback uses EventHandler
+    {
+      code: `
+        import { EventHandler, EventData } from '@fluentui/react-utilities';
+        export type OnSomeEventData = EventData<'focus', React.FocusEvent<HTMLElement>> & {
+          open: boolean,
+        };
+        export type SomeProps = {
+          onSomeEvent?: EventHandler<OnSomeEventData>,
+        };
+      `,
+      filename: 'src/components/SomeComponent/SomeComponent.type.ts',
+    },
+    // Valid when prop is TSIntersectionType and the callback uses EventHandler
+    {
+      code: `
+        import { EventHandler, EventData } from '@fluentui/react-utilities';
+        export type OnSomeEventData = EventData<'focus', React.FocusEvent<HTMLElement>> & {
+          open: boolean,
+        };
+        export type SomeProps = SomeType & {
+          onSomeEvent?: EventHandler<OnSomeEventData>,
+        };
+      `,
+      filename: 'src/components/SomeComponent/SomeComponent.type.ts',
+    },
+    // Valid when prop has a function that is not callback, and not using EventHandler type
+    {
+      code: `
+        export type SomeProps = {
+          someFunction?: EventHandler<OnSomeEventData>,
```

Comment:
this doesn't seem to test what the comment actually says.

something like following is more appropriate:

```
type SomeArgBag = {one:number};

        export type SomeProps = {
          someFunction?: (args:SomeArgBag) => void,
        };
```


🚨 also if type annotation is a Object literal it will explode  ( it should pass )


```ts
export type SomeProps = {
   someFunction?: (args: {one:string}) => void
};
```

⬇⬇⬇

<img width="576" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/2e9636f6-20c7-4831-87fa-d20204ad7562">